pkg/plugin/stream_reader.go

```diff
@@ -0,0 +1,55 @@
+package plugin
+
+import (
+	"bytes"
+	"io"
+)
+
+// ReceiveFunc is a function that either returns a slice
+// of bytes OR an error. Returning an io.EOF means there
+// is no more data to be read; any other error is considered
+// an actual error.
+type ReceiveFunc func() ([]byte, error)
+
+type CloseSendFunc func() error
+
+type StreamReadCloser struct {
+	buf     *bytes.Buffer
+	receive ReceiveFunc
+	close   CloseSendFunc
+}
+
+func (s *StreamReadCloser) Read(p []byte) (n int, err error) {
```

Comment:
We should get @davecheney to look at this too

Reply:
@davecheney could you take a look at this type if you get a sec? I can provide context if helpful